### PR TITLE
Release 11.6.5.3

### PR DESCRIPTION
**Pull request by krzychu124**
Because of mess with already merged stuff and incompatibilities with other mods it will be released to `Test` and `Stable` at the same time to bring back ability to swap between `Stable` <-> `Test`

Cherry-picked commits to build new release version, nothing to review.

- [x] What's new.txt (Test) needs to be updated to include changes between `11.6.5.2-hotfix-1` (what's already in the commits)
- [x] What's new.txt - short recap for `Stable` changelog - changes between `11.6.5.2-hotfix-1` and current Stable `11.6.4.8`
- [x] Cherry-picked API changes from #1577 
- [x] Readme.md/Changelog.md update?

Minor fixes after final tests:
- [x] Fixed `InvalidCastException` under certain condition prevented save/load of options, could mistakenly show message about path-finding update - airplane despawn
- [x] Fixed missing scrollbar at `Keybinds` tab
- [x] Corrected scope of DLS slider availability, improved visuals of read-only dropdown and slider
- [x] Fixed issue with misleading warning when changing `Road Signs` theme in main menu (no need to warn - `GlobalConfig` option)

ok... a lot of changes, actually
https://github.com/CitiesSkylinesMods/TMPE/compare/11.6.4.8...release/11.6.5.3

image

Link for testing [11.6.5.3 Stable.zip](https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=release/11.6.5.3)

### CODE REVIEW

**Comment by krzychu124** [comment]
There is a conflict... I'll fix it after release because I need last commit without changes from `master` to create a tag

File: TLM/TLM/Resources/whats_new.txt [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+[Link] tmpe-v11653-test-22052022
+[Mod] Stable version released
+[New] Vietnamese Translation (DucAnhLam)
+[Updated] Translations update #1551 (aubergine18)
+[Fixed] Regression with version naming #1568 (krzychu124)
+[/Version]
+
+[Version] 11.6.5.3
+[Released] May 22nd 2022
+[Stable]
```
Comment (aubergine10) [inline TLM/TLM/Resources/whats_new.txt:12]: This is going to be confusing to read - we've got 11.6.5.3 twice?

Also where did 11.6.5.2-hotfix-1 go (see PR https://github.com/CitiesSkylinesMods/TMPE/pull/1575)?

Suggestion: Leave the 11.6.5.2-hotfix-1 stuff as is, add a new 11.6.5.3 which is just hte translation stuff.

**Review by aubergine10 — CHANGES_REQUESTED** [review]

File: TLM/TLM/Resources/whats_new.txt (hunk 1 above)
Comment (aubergine10) [reply TLM/TLM/Resources/whats_new.txt:12]: Also, should we rename [11.6.5.2-hotfix-1 milestone](https://github.com/CitiesSkylinesMods/TMPE/milestone/66) to just be 11.6.5.3? If so is there other stuff that needs listing in the `whats_new.txt` and `CHANGELOG.md`?

Comment (krzychu124) [reply TLM/TLM/Resources/whats_new.txt:12]: > This is going to be confusing to read - we've got 11.6.5.3 twice?

No idea how we handle Test changelog when Stable is released at the same time. You leave Stable changelog only?

For hotfix1 it was just one issue/PR theoretically we can clean up and merge, it was already released so no one will notice version change 😂

Comment (krzychu124) [reply TLM/TLM/Resources/whats_new.txt:12]: eh,, nvm, I didn't notice `Closed` issues, but that doesn't matter, we can merge that

File: TLM/TLM/Lifecycle/TMPELifecycle.cs [hunk 2]
```diff
@@ -70,7 +70,18 @@ public class TMPELifecycle : MonoBehaviour {
         public bool MayPublishSegmentChanges()
             => InGameOrEditor() && !Instance.Deserializing;
 
-        public static AppMode? AppMode => SimulationManager.instance.m_ManagersWrapper.loading?.currentMode;
+        public static AppMode? AppMode {
```
[the remark this answers is not in the document]
Comment (kianzarrin) [reply TLM/TLM/Lifecycle/TMPELifecycle.cs:73]: shooting in the dark solution 1:
```suggestions
 => SimulationManager.instance?.m_ManagersWrapper?.loading?.currentMode;
```

shooting in the dark solution 2:
```suggestions
                try {
                    return SimulationManager.instance.m_ManagersWrapper.loading?.currentMode;
                }
                catch {
                    // unexpected error occurred.
                    return null;
                }
```

File: TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs [hunk 3]
```diff
@@ -64,10 +66,11 @@ public abstract class SerializableUIOptionBase<TVal, TUI, TComponent> : ILegacyS
 
         /// <summary>Gets or sets the value of the field this option represents.</summary>
         public virtual TVal Value {
-            get => _fieldInfo != null ? (TVal)_fieldInfo.GetValue(null) : _value;
+            get => _fieldInfo != null ? (TVal)Convert.ChangeType(_fieldInfo.GetValue(null), typeof(TVal)) : _value;
```
Comment (kianzarrin) [inline TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs:69]: maybe `Type.GetTypeCode()` could be be more efficient in avoiding heap allocation?

File: TLM/TLM/UI/Helpers/SliderOption.cs [hunk 4]
```diff
@@ -120,8 +120,11 @@ public byte FloatToByte(float val)
 
             Log._Debug($"SliderOption.UpdateReadOnly() - `{FieldName}` is {(readOnly ? "read-only" : "writeable")}");
 
+            _ui.isInteractive = !readOnly;
             _ui.thumbObject.isInteractive = !readOnly;
             _ui.thumbObject.opacity = readOnly ? 0.3f : 1f;
+            // parent is UIPanel containing text label and slider
+            _ui.parent.Find<UILabel>("Label").opacity = readOnly ? 0.3f : 1f;
```
Comment (kianzarrin) [inline TLM/TLM/UI/Helpers/SliderOption.cs:127]: `_sliderLabel` caches the label :)

File: TLM/TLM/Lifecycle/TMPELifecycle.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (krzychu124) [reply TLM/TLM/Lifecycle/TMPELifecycle.cs:73]: None of them will work. I wasn't shooting in the dark. I've had debugger attached all the time.

image

`m_properties` can be `null` on exit to main menu so comment it correct. I don't won't to copy paste whole code to just fix it. I'm 99% sure it shouldn't be used in this case anyways (outside of "modes" listed in the `AppMode` enum)

File: TLM/TLM/State/OptionsTabs/GeneralTab_LocalisationGroup.cs [hunk 5]
```diff
@@ -151,7 +151,12 @@ public static class GeneralTab_LocalisationGroup {
...
 
         private static void OnRoadSignsThemeChanged(int newThemeIndex) {
-            if (!Options.IsGameLoaded()) {
+            if (!Options.IsGameLoaded(false)) {
+                // update global config only, skip reloading theme
+                var selectedTheme = RoadSignThemeManager.Instance.ThemeNames[newThemeIndex];
+                Main config = GlobalConfig.Instance.Main;
+                config.RoadSignTheme = selectedTheme;
+                GlobalConfig.WriteConfig();
                 return;
```
Comment (kianzarrin) [inline TLM/TLM/State/OptionsTabs/GeneralTab_LocalisationGroup.cs:160]: this is duplicate code. instead you can guard the `switch` with `if`:
```cs
 if (Options.IsGameLoaded(false)) {
    // update global config only, skip reloading theme
    switch ( ... ) {
        ...
    }
}
```

Comment (krzychu124) [reply TLM/TLM/State/OptionsTabs/GeneralTab_LocalisationGroup.cs:160]: I'll leave that like it is and I see more bugs/issues/logic gaps with theme selection or forcing specific theme just by "use MPH" checkbox value above.

**Pull request merged**